### PR TITLE
feat: Implementation of a point surface

### PR DESCRIPTION
**Pull request by andiwand**
WIPWIP

### CODE REVIEW

File: Core/include/Acts/Surfaces/PointBounds.hpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#include <vector>
+
+namespace Acts {
+
+/// @class PointBounds
+///
+/// Bounds for a PointSurface.
+class PointBounds : public SurfaceBounds {
+ public:
+  enum BoundValues : int { eR = 0, eSize = 2 };
```
Comment (asalzburger) [inline Core/include/Acts/Surfaces/PointBounds.hpp:27]: I don't understand eSize ==2 here? Shouldn't that be `eSize==1` ?

File: Core/include/Acts/Surfaces/PointSurface.hpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+class PointBounds;
+class DetectorElementBase;
+class SurfaceBounds;
+
+/// @class PointSurface
+///
+/// Base class for a point surfaces in the TrackingGeometry to allow propagation
+/// to the closest point of approach of a 3D point. It inherits from Surface.
+///
+/// @note It leaves the type() method virtual, so it can not be instantiated
```
Comment (asalzburger) [inline Core/include/Acts/Surfaces/PointSurface.hpp:38]: Is this a leftover comment from the LineSurface?

File: Core/include/Acts/Surfaces/PointBounds.hpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+#include <array>
+#include <iosfwd>
+#include <stdexcept>
+#include <vector>
+
+namespace Acts {
+
+/// @class PointBounds
+///
```
Comment (asalzburger) [inline Core/include/Acts/Surfaces/PointBounds.hpp:23]: Can you add what the local position for a `PointSurface` means?

**Review by asalzburger — CHANGES_REQUESTED** [review]
Can you add a UnitTest for this ?

**Comment by andiwand** [comment]
> Can you add a UnitTest for this ?

sure! this is actually just a copy paste and some small adoptions from the `LineSurface` right now. will continue working on this when there is more time

**Comment by andiwand** [comment]
closing for now as I do not have time to work on that

**Pull request closed without merging**